plugins/modules/aci_interface_policy_storm_control.py

```diff
@@ -0,0 +1,433 @@
+        all_types_configuration=dict(type="dict", options=storm_control_policy_rate_spec(), aliases=["all_types"]),
+        broadcast_configuration=dict(type="dict", options=storm_control_policy_rate_spec(), aliases=["broadcast"]),
+        multicast_configuration=dict(type="dict", options=storm_control_policy_rate_spec(), aliases=["multicast"]),
+        unicast_configuration=dict(type="dict", options=storm_control_policy_rate_spec(), aliases=["unicast"]),
+        storm_control_action=dict(type="str", choices=["drop", "shutdown"]),
+        storm_control_soak_action=dict(type="int"),
+    )
+
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True,
+        required_if=[
+            ["state", "absent", ["storm_control_policy"]],
+            ["state", "present", ["storm_control_policy"]],
+        ],
+        mutually_exclusive=[
+            ("all_types_configuration", "broadcast_configuration"),
+            ("all_types_configuration", "multicast_configuration"),
+            ("all_types_configuration", "unicast_configuration"),
+        ],
+    )
+
+    aci = ACIModule(module)
+
+    storm_control_policy = module.params.get("storm_control_policy")
+    description = module.params.get("description")
+    state = module.params.get("state")
+    name_alias = module.params.get("name_alias")
+    storm_control_action = module.params.get("storm_control_action")
+    storm_control_soak_action = module.params.get("storm_control_soak_action")
+    all_types_configuration = module.params.get("all_types_configuration")
+    broadcast_configuration = module.params.get("broadcast_configuration")
+    multicast_configuration = module.params.get("multicast_configuration")
+    unicast_configuration = module.params.get("unicast_configuration")
+
+    rates_input = {}
+
+    def get_rates_configuration(configuration, percentage, pps, burst_percentage, burst_pps):
+        if configuration is None:
+            return {}
+        rate = configuration.get("rate")
+        burst_rate = configuration.get("burst_rate")
+        rate_type = configuration.get("rate_type")
+
+        if rate_type == "percentage":
+            for rate_name, rate_value in dict(rate=rate, burst_rate=burst_rate).items():
+                if rate_value is None or not (0 <= float(rate_value) <= 100):
+                    module.fail_json(
+                        msg="If argument rate_type is percentage, the {0} needs to be a value between 0 and 100 inclusive, got {1}".format(
+                            rate_name,
+                            rate_value,
+                        )
+                    )
+            return {
+                percentage: "{0:.6f}".format(float(rate)),
+                pps: "unspecified",
+                burst_percentage: "{0:.6f}".format(float(burst_rate)),
+                burst_pps: "unspecified",
+            }
+        elif rate_type == "pps":
+            return {pps: rate, burst_pps: burst_rate}
```

Comment:
Should we move the function outside of main function?

Reply:
for this, I cannot be sure. If we move this function outside of the main one, it will be end up in the aci.py file. And I feel like the aci.py has become the file where we put all of our module-specific outside functions and specs which leads to a lesser readable file the more we create new modules. Unless we create a new file with the purpose of stacking all of these specific functions and specs, I don't know which option is best.

Reply:
moved it!